backend/src/controllers/auth.py

```
@@ -0,0 +1,46 @@
+    payload.update({"exp":expiry_time})
+    access_token=jwt.encode(payload,SECRET_KEY,algorithm="HS256")   
+    print("AccessToken Giving:",access_token) 
+    return access_token
+
+#NOTE:So JWTs are passed around and stored as strings — nothing fancy. having three parts header.payload.signature
+def validate_token(access_token:str):
+    try:
+        payload=jwt.decode(access_token,SECRET_KEY,algorithms=os.getenv("ALGORITHM"))
+        return payload.get("subject") #ye data jis format me bheja se vaha se aayega {"subject":{"username":"aryan","id":"12324354657"}}        
+    except ExpiredSignatureError:
+        raise HTTPException(status_code=401, detail="Token has expired!Login Again")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)